libs/island-ui/contentful/src/lib/richTextRendering.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { FC, ReactNode, Fragment } from 'react'
+import {
+  Document,
+  Block,
+  Inline,
+  BLOCKS,
+  INLINES,
+} from '@contentful/rich-text-types'
+import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
+import Image from './Image/Image'
+import FaqList from './FaqList/FaqList'
+import { Slice } from '@island.is/api/schema'
+import { Statistics } from './Statistics/Statistics'
+import Hyperlink from './Hyperlink/Hyperlink'
+import {
+  Typography,
+  Blockquote,
+  Box,
+  TypographyProps,
+} from '@island.is/island-ui/core'
+import ProcessEntry from './ProcessEntry/ProcessEntry'
+import EmbeddedVideo from './EmbeddedVideo/EmbeddedVideo'
+import StaticHtml from './StaticHtml/StaticHtml'
+
+export interface RenderNode {
+  [k: string]: (node: Block | Inline, children: ReactNode) => ReactNode
+}
+
+interface RenderConfig {
+  renderComponent?: (slice: Slice, renderNode: RenderNode) => ReactNode
+  renderWrapper?: (slice: Slice, children: ReactNode) => ReactNode
+  renderPadding?: (top: Slice, bottom: Slice) => ReactNode
+  renderNode?: RenderNode
+  htmlClassName?: string
+}
+
+export const defaultRenderComponent = (
+  slice: Slice,
+  renderNode: RenderNode,
+  htmlClassName?: string,
+): ReactNode => {
+  switch (slice.__typename) {
+    case 'Html':
+      return renderHtml(slice.document, {
+        className: htmlClassName,
+        renderNode,
+      })
+
+    case 'FaqList':
+      return <FaqList {...slice} />
+
+    case 'Statistics':
+      return <Statistics {...slice} />
+
+    case 'Image':
+      return <Image type="apiImage" image={slice} />
+
+    case 'ProcessEntry':
+      return <ProcessEntry {...slice} />
+
+    case 'EmbeddedVideo':
+      return <EmbeddedVideo {...slice} />
+
+    default:
+      return null
+  }
+}
+
+export const defaultRenderWrapper = (
+  _slice: Slice,
+  children: ReactNode,
+): ReactNode => {
+  // TODO: when the grid supports it, we'll create a wrapper here to indent
+  // left and/or right side when needed
+  return children
+}
+
+const typography = (
+  variant: TypographyProps['variant'] & TypographyProps['as'],
+) => (_: Block, children: ReactNode) => (
+  <Typography variant={variant} as={variant}>
+    {['h2', 'h3'].includes(variant) ? (
+      // TODO: stop this data-sidebar-link madness
+      <span data-sidebar-link={String(children)}>{children}</span>
```

Comment:
will h2 and h3 be rendered as span data-sidebar-link everywhere?

Reply:
Yes, just like it is currently on the master branch. The sidebar depends on this for generating navigation for the page